experimental/GraphicalModels/src/PhylogeneticModels.jl

```diff
@@ -0,0 +1,271 @@
+struct PhylogeneticModel
+  n_states::Int
+  graph::Graph{Directed}
+  prob_ring::MPolyRing{QQFieldElem}
+  fourier_ring::MPolyRing{QQFieldElem}
+  trans_matrices::Dict{Edge, MatElem{QQMPolyRingElem}}
+  fourier_params::Dict{Edge, Vector{QQMPolyRingElem}}
+  group::Vector{Vector{Int64}}
+end
+
+struct FourierPhylogeneticModel
+  pm::PhylogeneticModel
+  fourier_ring::MPolyRing{QQFieldElem}
+  fourier_params::Dict{Edge, Vector{QQMPolyRingElem}}
+  group::Vector{Vector{Int64}}
+end
+
+
+# calling elements of the structure
+@doc raw"""
+    graph(pm::PhylogeneticModel)
+
+Return the graph of a PhylogeneticModel `pm`.
+"""
+graph(pm::PhylogeneticModel) = pm.graph
+
+@doc raw"""
+    transition_matrices(pm::PhylogeneticModel)
+
+Returns a dictionary between the edges of the tree specifying a PhylogeneticModel `pm` and attached transition matrices.
+"""
+transition_matrices(pm::PhylogeneticModel) = pm.trans_matrices
+
+@doc raw"""
+    number_states(pm::PhylogeneticModel)
+
+Return the number of states of the PhylogeneticModel `pm`.
+"""
+number_states(pm::PhylogeneticModel) = pm.n_states
+
+@doc raw"""
+    probabilities_ring(pm::PhylogeneticModel)
+
+Returns the ring of probability coordinates of the PhylogeneticModel `pm`.
+"""
+probabilities_ring(pm::PhylogeneticModel) = pm.prob_ring
+
+@doc raw"""
+    fourier_ring(pm::PhylogeneticModel)
+
+    Returns the ring of Fourier coordinates of the PhylogeneticModel `pm`.
+"""
+fourier_ring(pm::PhylogeneticModel) = pm.fourier_ring
+
+@doc raw"""
+    fourier_param(pm::PhylogeneticModel)
+
+Return the Fourier parameters of the PhylogeneticModel `pm` as a vector of eigenvalues of the transition matrices.
+"""
+fourier_parameters(pm::PhylogeneticModel) = pm.fourier_params
+
+group_model(pm::PhylogeneticModel) = pm.group
+
+
+#define group-based models
+@doc raw"""
+    jukes_cantor_model(graph::Graph{Directed})
+
+Creates a `PhylogeneticModel` based on `graph` whose transition matrices are Jukes Cantor matrices. 
+
+# Examples
+```jldoctest
+julia> pm = jukes_cantor_model(graph_from_edges(Directed,[[4,1],[4,2],[4,3]]));
+```
+We can recover the information of `pm`
+```jldoctest
+julia> graph(pm)
```

Comment:
This produces an error because `pm` is not defined. You need to inform Documenter that you wish to continue the previous example (see [here](https://documenter.juliadocs.org/stable/man/doctests/#Preserving-Definitions-Between-Blocks) for how to do that); or just merge them